README.md

```diff
@@ -1,289 +1,607 @@
+
+The collection of files within the directory **MAY** change between container launches.  The collection of files within the directory **SHOULD NOT** change during the lifetime of the container.
+
+## Example Directory Structure
+
+```plain
+$SERVICE_BINDING_ROOT
+├── account-database
+│   ├── kind
+│   ├── connection-count
+│   ├── uri
+│   ├── username
+│   └── password
+└── transaction-event-stream
+│   ├── kind
+│   ├── connection-count
+│   ├── uri
+│   ├── certificates
+│   └── private-key
+```
+
+# Service Binding
+
+A Service Binding describes the connection between a [Provisioned Service](#provisioned-service) and an [Application Projection](#application-projection).  It is codified as a concrete resource type.  Multiple Service Bindings can refer to the same service.  Multiple Service Bindings can refer to the same application.
+
+A Service Binding resource **MUST** define a `.spec.application` which is an `ObjectReference` to a `PodSpec`-able resource.  A Service Binding resource **MUST** define a `.spec.service` which is an `ObjectReference` to a Provisioned Service-able resource.  A Service Binding resource **MAY** define a `.spec.name` which is the name of the service when projected into the application.
+
+A Service Binding resource **MUST** define a `.status.conditions` which is an array of `Condition` objects.  A `Condition` object **MUST** define `type`, `status`, and `lastTransitionTime` entries.  At least one condition containing a `type` of `Ready` must be defined.  The `status` of the `Ready` condition **MUST** have a value of `True`, `False`, or `Unknown`.  The `lastTranstionTime` **MUST** contain the last time that the condition transitioned from one status to another.  A Service Binding resource **MAY** define `reason` and `message` entries to describe the last `status` transition.
+
+## Resource Type Schema
+
+```yaml
 apiVersion: service.binding/v1alpha1
 kind: ServiceBinding
 metadata:
-  name: example-service-binding
+  name:                 # string
 spec:
-  mountPathPrefix: "/var/bindings"
-  services:
-    - group: postgres.dev
-      kind: Service
-      name: global-user-db
-      version: v1beta1
-      id: postgres-global-user
-    - group: ibmcloud.ibm.com
-      version: v1alpha1
-      kind: Binding
-      name: coligo-service-binding
-      id: coligo-service-binding
-  application:
-    name: nodejs-rest-http-crud
-    group: apps
-    version: v1
-    resource: deployments
+  name:                 # string, optional, default: .metadata.name
+  kind:                 # string, optional
+  provider:             # string, optional
+
+  application:          # PodSpec-able resource ObjectReference
```

Comment:
We should add some way to specify which container(s) in a pod to bind into. By default we can bind into all containers, but since we're inherently dealing with sensitive values, that will not be desirable.

Reply:
Happy to have a concrete syntax recommended here.

Reply:
how are we dealing with injecting into resources that are owned by another controller, to avoid their reconciler from deleting the data?  We spoke before about strategies such as a mutation webhook, or creating the Secret(s) and letting the owner controller do the injection.

Reply:
Great eye @arthurdm.  We decided not to address the issue in the spec and leave it to the implementations to decide about their preferred strategy.  I'm thinking it might make sense to say that `application` SHOULD only refer to an "uncontrolled" resource (this removes the need completely), but then let implementations become more sophisticated and flexible by allowing references to "controlled" resources and doing the webhook trick (which has other downsides).

Thoughts on that path?

Reply:
Proposed a change in https://github.com/application-stacks/service-binding-specification/pull/48/files#r434131606

Reply:
I will take a stab at a proposal higher up in the text.

Reply:
Addressed via 511e8bf